Merge FieldTooltip and DetailPopover into single Popover component

marshmalien opened the pull request:
##### SUMMARY
We started adding “popovers” to the Detail views in settings, and since we already have a popover in FieldTooltip, I decided that it would be good to merge FieldTooltip and DetailPopover into a single component. I did a quick scan of our issues and moving from tooltips to popovers in this [PR](https://github.com/ansible/awx/pull/7905) should have resolved some of the following issues:

* Remote Archive SCM popover cleanup https://github.com/ansible/awx/issues/8140
* Form label popover alignment https://github.com/ansible/awx/issues/7556
* Clicking on popover selects the checkbox  https://github.com/ansible/awx/issues/6068
* Clicking on tooltip should open hoverer action (I think this is in reference to the form tooltips which are now popovers?) https://github.com/ansible/awx/issues/6067
* Hover-over tooltip actions become unusable when presenting (Tooltips are now popovers which should resolve this issue) https://github.com/ansible/awx/issues/6066
* Popover support to Variables Field https://github.com/ansible/awx/issues/6019

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - UI

<img width="811" alt="Screen Shot 2020-10-07 at 2 14 08 PM" src="https://user-images.githubusercontent.com/15881645/95370850-64490600-08a7-11eb-9ce3-c0b86c48661d.png">

marshmalien committed:
Merge FieldTooltip and DetailPopover into single Popover component

jakemcdermott commented on awx/ui_next/src/components/Popover/Popover.jsx:
```diff
@@ -0,0 +1,51 @@
+  margin: -(var(--pf-global--spacer--xs));
+`;
+
+function Popover({ content, header, id, maxWidth, ...rest }) {
```
:+1: :+1:

unlikelyzero commented:
@marshmalien I will be testing this tomorrow. I need to get jenkins back in action today to run the tests. If I can't, I'll just run the tests locally.

unlikelyzero commented:
@marshmalien this had some yolok failures. Let's sync on them to ensure they're not regressions

tiagodread commented:
We verified all these bugs but now this PR are introducing some a11y violations such as:
- `buttons have discernible text`
See: https://dequeuniversity.com/rules/axe/4.0/button-name?application=AxeChrome

Steps to reproduce:
1. Install Chrome axe extension
2. Run the analysis against the Job template form

This bug already exists on devel and would be easy to fix: 
- On the Job template Show changes option popover is displaying `If enabled, show the changes made by Ansible tasks, where supported. This is equivalent to Ansible&#x2019s --diff mode.` note the encoding problem here `Ansible&#x2019s`

marshmalien committed:
Add default aria label to Popover component

marshmalien committed:
Fix prop type errors in unit tests

marshmalien commented:
@tiagodread Pushed up changes to address these [issues](https://github.com/ansible/awx/pull/8346#issuecomment-712415079). 
* Added default aria label "More information" to Popover component
* Fixed up some unit test errors and removed the unicode apostrophe

unlikelyzero commented:
@marshmalien running through your changes with yolok. If they're 🟢  can we :mergeit: ?

marshmalien commented:
@unlikelyzero Yes, if it looks good then please merge away.
